rust/src/push/mod.rs

```diff
@@ -274,6 +279,8 @@ pub enum KnownCondition {
     #[serde(rename = "org.matrix.msc3772.relation_match")]
     RelationMatch {
         rel_type: Cow<'static, str>,
+        #[serde(skip_serializing_if = "Option::is_none", rename = "type")]
+        event_type_pattern: Option<Cow<'static, str>>,
```

Comment:
This should probably actually be in https://github.com/matrix-org/synapse/pull/13768, though I don't think we use it in practice?